File: packages/http-client-csharp/generator/Microsoft.TypeSpec.Generator.Input/src/InputTypes/InputOAuth2Auth.cs
// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.

using System.Collections.Generic;

namespace Microsoft.TypeSpec.Generator.Input
{
    public class InputOAuth2Auth
    {
        public InputOAuth2Auth(IReadOnlyCollection<InputOAuth2Flow> flows)
        {
            Flows = flows;
        }

        public IReadOnlyCollection<InputOAuth2Flow> Flows { get; }
    }
}

File: packages/http-client-csharp/emitter/src/type/input-oauth2-auth.ts
// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License. See License.txt in the project root for license information.

export interface InputOAuth2Auth {
  flows: InputOAuth2Flow[];
}

export interface InputOAuth2Flow {
  scopes?: string[];
  authorizationUrl?: string;
  tokenUrl?: string;
  refreshUrl?: string;
}
